src/bidiMapper/modules/script/PreloadScriptStorage.ts

```diff
@@ -68,9 +62,31 @@ export class PreloadScriptStorage {
   }
 
   /** Deletes all BiDi preload script entries that match the given filter. */
-  remove(filter?: PreloadScriptFilter) {
-    for (const preloadScript of this.find(filter)) {
-      this.#scripts.delete(preloadScript);
+  remove(id: string) {
+    const script = [...this.#scripts].find((script) => script.id === id);
+    if (script === undefined) {
+      throw new NoSuchScriptException(`No preload script with id '${id}'`);
+    }
+    this.#scripts.delete(script);
+  }
+
+  /** Gets the preload script with the given ID, if any, otherwise throws. */
+  getPreloadScript(id: string): PreloadScript {
+    const script = [...this.#scripts].find((script) => script.id === id);
+    if (script === undefined) {
+      throw new NoSuchScriptException(`No preload script with id '${id}'`);
+    }
+    return script;
+  }
+
+  onCdpTargetCreated(targetId: string, userContext: Browser.UserContext) {
+    for (const script of [...this.#scripts].filter((script) => {
+      if (!script.userContexts && !script.contexts) {
+        return true;
+      }
+      return script.userContexts?.includes(userContext);
```

Comment:
should not this use the same logic as the find with targetId method?

Reply:
That is impossible as at this point the Target is not in the list of the Target that the PreloadScript need to run.